frontend/src/components/AbstractModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { Dialog, DialogContent, DialogTitle, DialogActions, Typography, makeStyles, IconButton, Button } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import * as React from 'react';
+
+
+export type ModalType =  {
+    title: string;
+    description: string; 
+    confirmText: string;
+    cancelText: string;
+    isOpen: boolean;
+    handleConfirm: (e) => void;
+    handleCancel: (e) => void;
+};
+
+const useStyles = makeStyles(theme =>({
+    modal:{
+        position:'absolute',
+        top: theme.spacing(5),
+        padding: theme.spacing(2)
+    },
+    modalContent:{
+        textAlign: 'center'
+    },
+    modalAction:{
+        justifyContent: 'center'
+    },
+    modalTitle: {
+        position: 'absolute',
+        top: theme.spacing(2),
+        right: theme.spacing(0.5)
+    }
+}))
+
+
+export default function AbstractModal(props) {
+    const {modal, setModal} = props;
```

Comment:
So since you're not doing anything other than getting the attributes of `props`, you can use destructuring:
This:
`export default function AbstractModdal(props) {
    const {attribute1, attribute2} = props
}`
Becomes:
`export default function AbstractModal({attribute1, attribute2}) {
}`

Also, you should make sure to keep the attribute names consistent, as it helps in the compilation process.

Reply:
changed now no worries.